packages/integrations/markdoc/src/nodes/heading.ts

```diff
@@ -0,0 +1,36 @@
+import Markdoc, { type Schema } from '@markdoc/markdoc';
+import { getTextContent } from '../runtime.js';
+
+// Or replace this with your own function
+function generateId(attributes: Record<string, any>, textContent: string): string {
+	if (attributes.id && typeof attributes.id === 'string') {
+		return attributes.id;
+	}
+	return textContent.replace(/[?]/g, '').replace(/\s+/g, '-').toLowerCase();
+}
```

Comment:
For the default slug logic, should we use `github-slugger` so it's consistent with the `rehype-slug`? ([implementation](https://github.com/rehypejs/rehype-slug/blob/1c61632f1526c6b59c80530ed334d4a094c7a091/index.js#L33))